lib/api.js

```diff
@@ -0,0 +1,113 @@
+import base from 'taskcluster-base';
+import common from './common';
+import storage from './storage';
+import _ from 'lodash';
+
+/**
+ * Check to see if a resource an expiration has passed....
+ **/
+function hasExpired(expires) {
+  let currentTime = new Date();
+  let expireTime = new Date(expires);
+  if (currentTime > expireTime) {
+    return true;
+  }
+  return false;
+};
+
+/** API end-point for version v1/
+ *
+ * In this API implementation we shall assume the following context:
+ * {
+ *   publisher:      // publisher from base.Exchanges
+ * }
+ */
+var api = new base.API({
+  title:        "TaskCluster Secrets API Documentation",
+  description: [
+    "The secrets service, typically available at",
+    "`tools.taskcluster.net`, is responsible for managing",
+    "secure data via TaskCluster scopes."
+  ].join('\n')
+});
+
+// Export API
+module.exports = api;
+
+/** Define tasks */
+api.declare({
+  method:      'put',
+  route:       '/scope/:secretScope',
```

Comment:
I would recommend calling this `name` and using `/secret/:name(*)` as this allows slashes etc...

I suggest `/secret/` because the secret is the resource... not the scope... nit... but nice to try and be as restful as possible...